aiidalab/app.py

```diff
@@ -0,0 +1,647 @@
+    def title(self):
+        return self._get_from_metadata('title')
+
+    @property
+    def git_url(self):
+        """Provide explicit link to Git repository."""
+        if self._git_url is None:
+            return '-'
+        # else
+        return '<a href="{}">{}</a>'.format(self._git_url, self._git_url)
+
+    @property
+    def git_hidden_url(self):
+        """Provide a link to Git repository."""
+        if self._git_url is None:
+            return 'No Git url'
+        # else
+        return '<a href="{}"><button>Git URL</button></a>'.format(self._git_url)
+
+    @property
+    def more(self):
+        return """<a href=./single_app.ipynb?app={}>Manage App</a>""".format(self.name)
+
+    @property
+    def logo(self):
+        """Return logo object. Give the priority to the local version"""
+
+        # For some reason standard ipw.Image() app does not work properly.
+        res = ipw.HTML('<img src="./aiidalab_logo_v4.svg">', layout={'width': '100px', 'height': '100px'})
+
+        # Checking whether the 'logo' key is present in metadata dictionary.
+        if 'logo' not in self.metadata:
+            res.value = '<img src="./aiidalab_logo_v4.svg">'
+
+        # If 'logo' key is present and the app is installed.
+        elif self.is_installed():
+            res.value = '<img src="{}">'.format(path.join('..', self.name, self.metadata['logo']))
+
+        # If not installed, getting file from the remote git repository.
+        else:
+            # Remove .git if present.
+            html_link = os.path.splitext(self._git_url)[0]
+
+            # We expect it to always be a git repository
+            html_link += '/master/' + self.metadata['logo']
+            if 'github.com' in html_link:
+                html_link = html_link.replace('github.com', 'raw.githubusercontent.com')
+                if html_link.endswith('.svg'):
+                    html_link += '?sanitize=true'
+            res.value = '<img src="{}">'.format(html_link)
+
+        return res
+
+    @property
+    def repo(self):
+        """Returns Git repository."""
+        if not hasattr(self, '_repo'):
+            if self.is_installed():
+                self._repo = Repo(self._get_appdir())
+            else:
+                raise AppNotInstalledException("The app is not installed")
+        return self._repo
+
+    @property
+    def update_info(self):
+        """Update app info."""
+        if not self.has_git_repo():
+            return """<font color="#D8000C"><i class='fa fa-times-circle'></i> Not a Git Repo</font>"""
+        if not self._git_url:
+            return """<font color="#D8000C"><i class='fa fa-times-circle'></i> No remote URL</font>"""
+        if self.git_update_available():
+            return """<font color="#9F6000"><i class='fa fa-warning'></i> Update Available</font>"""
+        return """<font color="#270"><i class='fa fa-check'></i> Latest Version</font>"""
+
+    def render(self):
+        """"Display the app."""
```

Comment:
It's showing the app widget, not the app, right? I think that should be made clear in the name of this function and its doc-string. Maybe `render_start_widget` or something is more appropriate?

Reply:
Renamed to `render_app_manager_widget`